doc/CMakeLists.txt

```diff
@@ -264,7 +280,7 @@ add_custom_target(
   PYTHONPATH=${ZEPHYR_BASE}/scripts/dts/python-devicetree/src${SEP}$ENV{PYTHONPATH}
   ZEPHYR_BASE=${ZEPHYR_BASE}
   GEN_DEVICETREE_REST_ZEPHYR_DOCSET=${GEN_DEVICETREE_REST_ZEPHYR_DOCSET}
-  ${PYTHON_EXECUTABLE} ${GEN_DEVICETREE_REST_SCRIPT}
+  ${PYTHON_EXECUTABLE} ${PYTHONOPTS} ${GEN_DEVICETREE_REST_SCRIPT}
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/issues/34076 may be fixed by just adjusting the encoding used by the DTS generator script.

Reply:
> #34076 may be fixed by just adjusting the encoding used by the DTS generator script.

If we start that game of whackamole then we'll never stop; I'm trying to find a solution that doesn't require us to take special care everywhere.

Reply:
I think it's the script the responsible to handle the encoding and not CMake via Python option. As mentioned, Sphinx takes care of doing that when dealing with project sources, we should do the same. I think current CMake is convoluted enough to not add more flags that, by the way, are not available on all Python versions we support.

Reply:
Another point: if any of the current scripts ends up being proper Sphinx extensions, we can't rely on `-X utf8`.

Reply:
I think it's fair for Zephyr scripts to expect to be run in a UTF-8 compatible way. A future question about converting to sphinx extensions seems like a hypothetical.

Reply:
Since the script knows that the files going to be handled are UTF-8 here, it's the responsible to read/write with the appropriate encoding. It's annoying that Windows doesn't default to utf-8...

There's another solution (not tested) that may not depend on having Python >= 3.7: set `PYTHONUTF8=1` on CMake

Reply:
To me this boils down to 2 things:

- the reporter confirms this is a fix
- we already have a precedent set in the Zephyr build system's python.cmake file that we are willing to add in Windows-specific workarounds

So this is the fix I would suggest. Feel free to file your own patch of course!